common/deftype/deftype.cpp

```diff
@@ -1623,12 +1651,23 @@ extern DEFTYPE_API ITypeInfo *makeIntType(int len, bool isSigned)
     if (len <= 0 || len > 8)
         return NULL;
 
-    CriticalBlock procedure(*typeCS);
-    CIntTypeInfo *ret = itt[isSigned][len-1];
-    if (ret==NULL)
-        ret = itt[isSigned][len-1] = new CIntTypeInfo(len, isSigned);
-    ::Link(ret);
-    return ret;
+    std::atomic<CIntTypeInfo*>& singleton = itt[isSigned][len-1];
```

Comment:
reuse: simpler if this also used querySingleton(), or reused the function above.  (One reason why passing in the function to create the object is a good idea).
Same comment goes for all the other functions for objects with non-default constructors.

Reply:
Agreed, will do.